python/py-pip/Portfile

```diff
@@ -76,13 +76,20 @@ if {${name} ne ${subport}} {
             xinstall -m 644 {*}[glob -types f ${worksrcpath}/docs/reference/*] ${destroot}${prefix}/share/doc/${subport}/reference
         }
 
-        delete ${destroot}${prefix}/bin/pip[string range ${python.version} 0 end-1]-${python.branch}
-        delete ${destroot}${prefix}/bin/pip${python.branch}-${python.branch}
+        delete [glob -directory ${destroot}${prefix}/bin pip[string index ${python.branch} 0]-*]
+        delete [glob -directory ${destroot}${prefix}/bin pip${python.branch}-*]
+    }
+
+
+    if {[string match pypy* ${subport}]} {
```

Comment:
If we could avoid this kind of special casing in ports it would be much, much preferable.

Reply:
I agree, but this stems from `py-pip` using `pip${python.version}` for it's selectors. If we had `${python.version}` as `py37` for PyPy 3.7, that would lead to `pippy37`. In the current implementation, it leads to both selectors being `pip37`. Hopefully, this won't occur in too many other ports.

Reply:
Maybe there's a case for standardising how those are generated?

Reply:
> Maybe there's a case for standardising how those are generated?

Sure; can I do so without breaking backwards compatibility? Say we rename the `pip38` group to `py38-pip`, how would that be handled in practice? Does some logic for handling that already exist, or would we need to add it?